### DISCUSSION TITLE
Tutorial topics suggestion

### DISCUSSION THREAD

**Discussion by HamedFathi** [ged-odoo/blockdom#30] [General]
Hi,

I am not sure is a meaningful request or not! but this project is special because
1. It is one of the best VDOM out there 
2. About VDOM, I have never seen such a great tutorial that you have written (a framework implementation)
3. Not only is it a unique library but it is also a place where you can learn a lot of information and details about behind the scenes design of SPA frameworks.

So, I want to ask for more tutorials (if possible please)

A. `blockdom` and WebComponents (Is it possible to have web components based on `blockdom`?)
B. `blockdom` and one way/two ways binding (Angular-like binding or ... with `blockdom` )
C. `blockdom` and JSX

**Comment by ged-odoo** [comment]
Hey @HamedFathi Thanks for your interest, it means a lot to me :)

Yes, I am happy about this project, it is obviously one of the goal of the project to be fast. But clearly, I learned a lot working on this and on the tutorial. It is kind of an experimental playground so I can toy with ideas and learn interesting way to combine them.

I am currently working on extending `tomato` to be asynchronous. My goal is to support components looking like this:
```js
async function MyComponent() {
  let [state, setState] = useState(0);
  await someAsyncWork();
  return props => html`...`
}
```
But it is a pretty big topic, so I am not sure something interesting/simple will come out of it.

So, yeah, excited to try other ideas. I'll think about your suggestions, and when I get the time, I'll see what we can do.  Maybe this project will slowly transform into a large list of resources to learn about javascript and frameworks, who knows?

Don't hesitate to add other tutorial ideas if you want! (however, obviously, I cannot promise anything!)

**Reply by HamedFathi** [reply]
@ged-odoo you did a great job!

Honestly, I had an idea to create my own framework mix of [FAST](https://www.fast.design/docs/fast-element/getting-started) and [Aurelia](http://aurelia.io/docs/) especially [Aurelia v2](https://docs.aurelia.io/) ideas but without decorators and class.

Getting to know you and this project gave me hope.

Thanks for everything.

**Reply by HamedFathi** [reply]
For example, a generic html template is really interesting (from FAST)

```js
const template = html<NameTag>`
  <div class="header">
    <h3>${x => x.greeting.toUpperCase()}</h3>
    <h4>my name is</h4>
  </div>

  <div class="body">TODO: Name Here</div>

  <div class="footer"></div>
`;
```

**Comment by ged-odoo** [comment]
also, another topic I would like to attempt someday is using mobx with blockdom
